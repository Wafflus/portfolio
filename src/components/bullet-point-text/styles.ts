import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;