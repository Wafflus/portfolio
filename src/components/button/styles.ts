import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;