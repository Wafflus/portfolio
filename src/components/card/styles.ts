import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;