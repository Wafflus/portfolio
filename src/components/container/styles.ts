import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;