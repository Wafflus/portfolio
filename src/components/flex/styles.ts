import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;