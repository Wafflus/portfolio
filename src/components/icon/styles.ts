import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;