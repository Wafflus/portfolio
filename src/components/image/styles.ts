import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;