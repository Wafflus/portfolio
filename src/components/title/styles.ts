import styled from "styled-components";

export const TitleStyled = styled.h2`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: bold;
    letter-spacing: 0.1rem;
    margin-bottom: 32px;
    margin-top: 32px;
    position: relative;
    text-align: center;

    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 3rem;
            margin-bottom: 64px;
            margin-top: 0;
        `
    )}
`;

export const LineStyled = styled.div`
    border-bottom: 2px solid ${({ theme }) => theme.data.text.defaultColor};
    left: calc(50% - 150px);
    position: absolute;
    top: 50%;
    width: 300px;
    z-index: -10;

    &::before,
    &::after {
        background-color: ${({ theme }) => theme.data.text.defaultColor};
        border-radius: 50%;
        bottom: -6px;
        content: '';
        height: 10px;
        position: absolute;
        width: 10px;
    }

    &::before {
        left: -4px;
    }

    &::after {
        right: -4px;
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            left: calc(50% - 200px);
            width: 400px;
        `
    )}
`;

export const SpanStyled = styled.span`
    background-color: ${({ theme }) => theme.data.lightColor};
    padding: 0px 32px;
`;