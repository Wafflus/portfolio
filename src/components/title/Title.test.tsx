import "jest-styled-components";
import { breakpoints, theme } from "styles";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { Title } from ".";

test('title renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Title>
                <>Title</>
            </Title>
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toBeInTheDocument();
});

test('should have different styles on large (lg) screens', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Title>
                <>Title</>
            </Title>
        </ThemeProvider>
    );

    const options = { media: `(min-width: ${breakpoints.lg}px)` };

    expect(container.firstChild).toHaveStyleRule('font-size', '3rem', options);
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '64px', options);
    expect(container.firstChild).toHaveStyleRule('margin-top', '0', options);
});

test('should have line under the title', () => {
    const title = render(
        <ThemeProvider theme={theme}>
            <Title>
                <>Title</>
            </Title>
        </ThemeProvider>
    );

    const line = title.getByRole('presentation').closest('div');

    expect(line).toBeInTheDocument();
});

test('line should have circle styles at the start (left) side', () => {
    const title = render(
        <ThemeProvider theme={theme}>
            <Title>
                <>Title</>
            </Title>
        </ThemeProvider>
    );

    const line = title.getByRole('presentation').closest('div');
    const options = { modifier: '::before' };

    expect(line).toHaveStyleRule('background-color', theme.data.text.defaultColor, options);
    expect(line).toHaveStyleRule('border-radius', '50%', options);
    expect(line).toHaveStyleRule('bottom', '-6px', options);
    expect(line).toHaveStyleRule('content', '\'\'', options);
    expect(line).toHaveStyleRule('height', '10px', options);
    expect(line).toHaveStyleRule('position', 'absolute', options);
    expect(line).toHaveStyleRule('width', '10px', options);
    expect(line).toHaveStyleRule('left', '-4px', options);
});

test('line should have circle styles at the end (right) side', () => {
    const title = render(
        <ThemeProvider theme={theme}>
            <Title>
                <>Title</>
            </Title>
        </ThemeProvider>
    );

    const line = title.getByRole('presentation').closest('div');
    const options = { modifier: '::after' };

    expect(line).toHaveStyleRule('background-color', theme.data.text.defaultColor, options);
    expect(line).toHaveStyleRule('border-radius', '50%', options);
    expect(line).toHaveStyleRule('bottom', '-6px', options);
    expect(line).toHaveStyleRule('content', '\'\'', options);
    expect(line).toHaveStyleRule('height', '10px', options);
    expect(line).toHaveStyleRule('position', 'absolute', options);
    expect(line).toHaveStyleRule('width', '10px', options);
    expect(line).toHaveStyleRule('right', '-4px', options);
});

test('line should have different styles on large (lg) screens', () => {
    const title = render(
        <ThemeProvider theme={theme}>
            <Title>
                <>Title</>
            </Title>
        </ThemeProvider>
    );

    const line = title.getByRole('presentation').closest('div');
    const options = { media: `(min-width: ${breakpoints.lg}px)` };

    expect(line).toHaveStyleRule('left', 'calc(50% - 200px)', options);
    expect(line).toHaveStyleRule('width', '400px', options);
});