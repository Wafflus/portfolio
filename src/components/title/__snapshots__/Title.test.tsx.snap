// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title renders 1`] = `
.c0 {
  color: #93855B;
  font-size: 1.5rem;
  font-variant: small-caps;
  font-weight: bold;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  margin-bottom: 32px;
  margin-top: 32px;
  position: relative;
  text-align: center;
}

.c2 {
  border-bottom: 2px solid #93855B;
  left: calc(50% - 150px);
  position: absolute;
  top: 50%;
  width: 300px;
  z-index: -10;
}

.c2::before,
.c2::after {
  background-color: #93855B;
  border-radius: 50%;
  bottom: -6px;
  content: '';
  height: 10px;
  position: absolute;
  width: 10px;
}

.c2::before {
  left: -4px;
}

.c2::after {
  right: -4px;
}

.c1 {
  background-color: #FFF5D6;
  padding: 0px 32px;
}

@media (min-width:992px) {
  .c0 {
    font-size: 3rem;
    margin-bottom: 64px;
    margin-top: 0;
  }
}

@media (min-width:992px) {
  .c2 {
    left: calc(50% - 200px);
    width: 400px;
  }
}

<h2
  class="c0"
>
  <span
    class="c1"
  >
    Title
  </span>
  <div
    class="c2"
    role="presentation"
  />
</h2>
`;
